compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -245,6 +245,14 @@ macro_rules! is_anon {
     };
 }
 
+macro_rules! if_eval_always {
```

Comment:
What effect does this have exactly?

Reply:
It expands to the provided item only if `eval_always` is specified for a query.

Reply:
It would be great to document this.